src/chexus/validators.py

```diff
@@ -154,6 +154,22 @@ def validate(self, node: Dataset | Group) -> Violation | None:
             return Violation(node.name)
 
 
+class float_dataset_units_check(float_dataset_units_missing):
```

Comment:
I don't think it is clean to derive `float_dataset_units_missing` here because it breaks the Liskov substitution principle (it doesn't cleanly check for absence of a unit).